# HG changeset patch
# User Rik <rik@octave.org>
# Date 1408555149 25200
#      Wed Aug 20 10:19:09 2014 -0700
# Node ID f707835af86783242964216c2ad65d0a8237952b
# Parent  c8240a60dd01140098ddc660aebf5293753163b3
Rewrite schur built-in test as a %!warning to stop failure during 'make check'
Some odd interaction allows the test to complet when run interactively, but
not when run during 'make check'
* schur.cc: Rewrite '%! fail' test as '%!warning'.

diff --git a/libinterp/corefcn/schur.cc b/libinterp/corefcn/schur.cc
--- a/libinterp/corefcn/schur.cc
+++ b/libinterp/corefcn/schur.cc
@@ -281,25 +281,23 @@ Riccati equations in control (see @code{
 %! [u, s] = schur (a);
 %! assert (u' * a * u, s, sqrt (eps));
 
 %!test
 %! a = single ([1, 2, 3; 4, 5, 9; 7, 8, 6]);
 %! [u, s] = schur (a);
 %! assert (u' * a * u, s, sqrt (eps ("single")));
 
-%!test
-%! fail ('schur ([1, 2; 3, 4], "bad_opt")',
-%!       "warning", "incorrect ordered schur argument");
-
 %!error schur ()
 %!error schur (1,2,3)
 %!error [a,b,c] = schur (1)
 %!error <argument must be a square matrix> schur ([1, 2, 3; 4, 5, 6])
 %!error <wrong type argument 'cell'> schur ({1})
+%!warning <incorrect ordered schur argument> schur ([1, 2; 3, 4], "bad_opt");
+
 */
 
 DEFUN (rsf2csf, args, nargout,
        "-*- texinfo -*-\n\
 @deftypefn {Function File} {[@var{U}, @var{T}] =} rsf2csf (@var{UR}, @var{TR})\n\
 Convert a real, upper quasi-triangular Schur@tie{}form @var{TR} to a complex,\n\
 upper triangular Schur@tie{}form @var{T}.\n\
 \n\
